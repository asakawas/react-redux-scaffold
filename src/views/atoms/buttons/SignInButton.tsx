import React, { FunctionComponent } from 'react';
import SpacingButton, { SpacingButtonProps } from '~/views/atoms/buttons/SpacingButton';

/**
 * The type of props of SignInButton.
 */
export type SignInButtonProps = Readonly<
  Pick<SpacingButtonProps, 'ml' | 'disabled'> & {
    /** The event handler called when the button is clicked */
    onSignInButtonClick: SpacingButtonProps['onClick'];
  }
>;

const SignInButton: FunctionComponent<SignInButtonProps> = ({
  onSignInButtonClick,
  ml,
  disabled,
}) => (
  <SpacingButton
    variant="contained"
    aria-label="sign in button"
    color="primary"
    onClick={onSignInButtonClick}
    ml={ml}
    disabled={disabled}
  >
    Sign In
  </SpacingButton>
);

export default SignInButton;
