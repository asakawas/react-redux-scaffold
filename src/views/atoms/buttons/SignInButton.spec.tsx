import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SignInButton from './SignInButton';

describe('SignInButton', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(
      <SignInButton onSignInButtonClick={() => {}} ml={1} disabled={false} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("calls the passed handler when it's clicked if not disabled", () => {
    const handler = jest.fn();
    const renderResult = render(
      <SignInButton onSignInButtonClick={handler} ml={1} disabled={false} />,
    );

    const tree = renderResult.getByText('Sign In');
    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(1);

    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(2);
  });

  test("calls the passed handler when it's clicked if disabled", () => {
    const handler = jest.fn();
    const renderResult = render(<SignInButton onSignInButtonClick={handler} ml={1} disabled />);

    const tree = renderResult.getByText('Sign In');
    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(0);

    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(0);
  });

  test('has an attribute "ml" with its value set to the given number', () => {
    const renderResult = render(
      <SignInButton onSignInButtonClick={() => {}} ml={11} disabled={false} />,
    );

    const btn = renderResult.getByLabelText('sign in button');
    expect(btn).toHaveAttribute('ml', '11');
  });
});
