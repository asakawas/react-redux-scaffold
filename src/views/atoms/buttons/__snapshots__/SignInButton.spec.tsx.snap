// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInButton is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  margin-left: 8px;
}

<button
    aria-label="sign in button"
    class="MuiButtonBase-root MuiButton-root MuiButton-contained c0 MuiButton-containedPrimary"
    ml="1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Sign In
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
